src/client/scripts/game/chess/checkdetection.js

```diff
@@ -275,8 +275,8 @@ const checkdetection = (function(){
      */
     function addressExistingChecks (gamefile, legalMoves, royalCoords, selectedPieceCoords, color) {
         if (!gamefile.inCheck) return false; // Exit if not in check
-        const isOurCheck = color === gamefile.whosTurn;
-        if (!isOurCheck) return; // Our OPPONENT is in check, not us! Them being in check doesn't restrict our movement!
+        
+        if (!areWeInCheck(gamefile, color)) return; // Our OPPONENT is in check, not us! Them being in check doesn't restrict our movement!
```

Comment:
This line broke legal moves when you are in check:

<img width="1167" alt="Screen Shot 2024-08-07 at 3 29 24 PM" src="https://github.com/user-attachments/assets/f17921c6-5fd3-4ed7-a535-b1e87db89187">

I see the need though to differentiate our royals in check from any other that may be in check this turn. Because with multiple players, multiple colors can be in check at once, it doesn't necessarily need to be our turn for us to be in check.